sfxclient/multitokensink.go

```diff
@@ -0,0 +1,734 @@
+package sfxclient
+
+import (
+	"fmt"
+	"hash"
+	"hash/fnv"
+	"net/http"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/event"
+	"golang.org/x/net/context"
+)
+
+// ContextKey is a custom key type for context values
+type ContextKey string
+
+const (
+	// TokenCtxKey is a context key for tokens
+	TokenCtxKey ContextKey = TokenHeaderName
+)
+
+// dpMsg is the message object for datapoints
+type dpMsg struct {
+	token string
+	data  []*datapoint.Datapoint
+}
+
+// evMsg is the message object for events
+type evMsg struct {
+	token string
+	data  []*event.Event
+}
+
+type tokenStatus struct {
+	status string
+	token  string
+	val    int64
+}
+
+type asyncCounter struct {
+	name              string
+	val               int64
+	input             chan int64
+	stop              chan bool
+	requestDatapoints chan bool
+	returnDatapoints  chan []*datapoint.Datapoint
+	defaultDims       map[string]string
+	previousVal       []*datapoint.Datapoint
+}
+
+func (a *asyncCounter) Datapoints() (dps []*datapoint.Datapoint) {
+	var request bool
+	for {
+		select {
+		case dps = <-a.returnDatapoints:
+			if dps == nil {
+				dps = a.previousVal
+			}
+			return
+		default:
+			if !request {
+				a.requestDatapoints <- true
+				request = true
+			}
+		}
+	}
+}
+
+// AddDatapoints adds a tokenStatus object to the counter
+func (a *asyncCounter) Increment(inc int64) {
```

Comment:
doesn't look like gobuild was run on this, an exported Increment function would not be able to have an //AddDatapoints comment...

Reply:
It needs to be fixed, but the reason gobuild passed is because the struct the function is attached to isn't exported.